src/rgw/rgw_main.cc

```diff
@@ -202,15 +202,24 @@ int radosgw_Main(int argc, const char **argv)
     exit(0);
   }
 
-  // First, let's determine which frontends are configured.
   int flags = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
   global_pre_init(
     &defaults, args, CEPH_ENTITY_TYPE_CLIENT, CODE_ENVIRONMENT_DAEMON,
     flags);
 
+  // Now that we've determined which frontend(s) to use, continue with global
+  // initialization. Passing false as the final argument ensures that
+  // global_pre_init() is not invoked twice.
+  // claim the reference and release it after subsequent destructors have fired
+  auto cct = global_init(&defaults, args, CEPH_ENTITY_TYPE_CLIENT,
+			 CODE_ENVIRONMENT_DAEMON,
+			 flags, "rgw_data", false);
```

Comment:
@yehudasa a regression was reported in https://tracker.ceph.com/issues/44661 because we're now calling global_init() before this happens below:
```c++
      // If civetweb or beast is configured as a frontend, prevent global_init() from
      // dropping permissions by setting the appropriate flag.
      flags |= CINIT_FLAG_DEFER_DROP_PRIVILEGES;
```
our teuthology suite uses these privileged ports and hasn't seen any failures here, so I'm not sure where the gap is